.github/workflows/python-docs.yml

```diff
@@ -45,8 +44,8 @@ jobs:
 
           # Only try to commit if there exist changes
           git diff --quiet HEAD || {
-              git commit -m 'Python docs for '$this_sha
+              git commit -m 'Python docs for '$built_sha
               git push origin HEAD:gh-pages-bench
           }
 
-          git checkout 'HEAD@{2}'
+          git checkout $built_sha
```

Comment:
Previously if the diff was empty (and thus we did not create a commit), the reflog only had one entry: the checkout. This is more robust: we switch back to the specific SHA that the cleanup actions expect to exist.